index.js

```diff
@@ -1,254 +1,263 @@
-var async = require('async')
-var debug = require('debug')('systemic:index')
-var format = require('util').format
-var Toposort = require('toposort-class')
-var getProp = require('lodash.get')
-var setProp = require('lodash.set')
-var hasProp = require('lodash.has')
-var map = require('lodash.map')
-var find = require('lodash.find')
-var isFunction = require('lodash.isfunction')
-var toArray = require('lodash.toarray')
-var defaults = require('lodash.defaults')
-var assign = require('lodash.assign')
-var intersection = require('lodash.intersection')
-var requireAll = require('require-all')
-var Chance = require('chance')
+const async = require('async')
+const debug = require('debug')('systemic:index')
+const format = require('util').format
+const Toposort = require('toposort-class')
+const {
+    randomName,
+    isFunction,
+    arraysIntersection,
+    hasProp,
+    getProp,
+    setProp
+} = require('./utils')
+const requireAll = require('require-all')
 
-module.exports = function(_params) {
-
-    var params = assign({}, { name: new Chance().first() }, _params)
-    var definitions = {}
-    var currentDefinition
-    var running = false
-    var started
-    var defaultComponent = {
-        start: function(dependencies, cb) {
+module.exports = (_params) => {
+    const params = Object.assign({}, {name: randomName()}, _params)
+    let definitions = {}
+    let currentDefinition
+    let running = false
+    let started
+    let defaultComponent = {
+        start: (dependencies, cb) => {
             cb(null, dependencies)
         }
     }
 
-    function bootstrap(path) {
+    const bootstrap = (path) => {
```

Comment:
I've to disagree there xD. I think this topic is very opinated and we should discuss in a issue. I'm more used to arrow function syntax and I think arrow style is more concise. `function` keyword is more descriptive because you are defining it with the keyword, but on the other hand, you will have to deal with the `this` pain

On the other hand, I think the var -> const/let must be done asap because all the IDEs complain about that.

Reply:
`const name = () => {` is denser than `function name() {`, but less concise.

> concise: *giving a lot of information clearly and in a few words; brief but comprehensive.*
> dense: *closely compacted in substance.*

High density code is something we should be suspicious of since it frequently results in less readable code. We might feel proud when writing it, be enticed by shiny new features of ES6, or influenced by external/our IDE's linting rules, but in reality we be aspiring to write code which communicates well, and this is not necessarily the same thing.

As you've rightly pointed out, how well code communicates is subjective, but there are objective principles we can lean on.

* We have trained our brains to parse sentences by pattern matching individual words based on the shape of their letters. The more words, the more work our brain has to do to parse the sentence. The same is true for lines of code. `const name = (params) => {` has more words than `function name(params) {` (6 vs 4). 
* While a sentence can have multiple meanings our brain has to keep track of both. The lead term `const` leaves multiple options for what follows (a value or a function), whereas the lead term `function` is unambiguous.
* Putting the most salient detail at the beginning of a sentence aids skim reading. `function` is a highly salient detail, whereas `const` is largely irrelevant.
* We have trained our brains to recognise words by their shapes (at least in the west), but we are far less used to recognising individual characters or symbols, especially when they are similar. Which of the following is easier to read, d a f f o d i l or daffodil? `const name = (arg) => {` or worse `const name = () => {` ends with a sequence of characters / symbols which is far harder for your brain to parse than `function name() {`

Once you discount familiarity with different code styles `function name() {` is objectively more readable than `const name = () => {` hence why I prefer it.


*For the avoidance of doubt, I am in favour of arrow functions for callbacks, but suspicious of them when they are declared and assigned to constants.*